src/modules/displaymanager/main.py

```diff
@@ -847,7 +848,9 @@ def set_autologin(self, username, do_autologin, default_desktop_environment):
         self.config_load()
 
         de_command = default_desktop_environment.executable
-        if os.path.exists(self.os_path("usr/bin/gtkgreet")) and os.path.exists(self.os_path("usr/bin/cage")):
+        if os.path.exists(self.os_path("usr/bin/gtkgreet")) and os.path.exists(self.os_path("usr/bin/cage")) and os.path.exist(self.os_path(self.greeter_css_location)):
+            self.config_data['default_session']['command'] = "cage -d -s -- gtkgreet -s" + self.greeter_css_location
+        elif os.path.exists(self.os_path("usr/bin/gtkgreet")) and os.path.exists(self.os_path("usr/bin/cage")):
```

Comment:
This could be refactored to avoid the logic duplication.

Reply:
Please check new logic